src/api/classify_element.rs

```diff
@@ -318,46 +337,149 @@ mod tests {
-        assert_eq!(elem0_port0_output.len(), 1000);
+        let even_output: Vec<_> = even_collector_output.iter().collect();
+        assert_eq!(even_output.len(), 1000);
 
-        let elem0_port1_output: Vec<_> = elem0_port1_collector_output.iter().collect();
-        assert_eq!(elem0_port1_output.len(), 1000);
+        let odd_output: Vec<_> = odd_collector_output.iter().collect();
+        assert_eq!(odd_output.len(), 1000);
+    }
+
+    #[test]
+    fn one_fizz_buzz() {
+        let default_channel_size = 10;
+        let packet_generator = immediate_stream(0..=30);
+
+        let elem = ClassifyFizzBuzzElement { id: 0 };
+
+        let mut elem_link =
+            ClassifyElementLink::new(Box::new(packet_generator), elem, default_channel_size, 4);
+        let elem_drain = elem_link.consumer;
+
+        let (s3, other_output) = crossbeam_channel::unbounded();
+        let port3_collector =
+            ExhaustiveCollector::new(0, Box::new(elem_link.providers.pop().unwrap()), s3);
+
+        let (s2, buzz_output) = crossbeam_channel::unbounded();
+        let port2_collector =
+            ExhaustiveCollector::new(0, Box::new(elem_link.providers.pop().unwrap()), s2);
+
+        let (s1, fizz_output) = crossbeam_channel::unbounded();
+        let port1_collector =
+            ExhaustiveCollector::new(0, Box::new(elem_link.providers.pop().unwrap()), s1);
+
+        let (s0, fizz_buzz_output) = crossbeam_channel::unbounded();
+        let port0_collector =
+            ExhaustiveCollector::new(0, Box::new(elem_link.providers.pop().unwrap()), s0);
+
+        tokio::run(lazy(|| {
+            tokio::spawn(elem_drain);
+            tokio::spawn(port0_collector);
+            tokio::spawn(port1_collector);
+            tokio::spawn(port2_collector);
+            tokio::spawn(port3_collector);
+            Ok(())
+        }));
+
+        let actual_fizz_buzz = fizz_buzz_output.iter().collect::<Vec<i32>>();
+        let expected_fizz_buzz = vec![0, 15, 30];
+        assert_eq!(expected_fizz_buzz, actual_fizz_buzz);
+
+        let actual_fizz = fizz_output.iter().collect::<Vec<i32>>();
+        let expected_fizz = vec![3, 6, 9, 12, 18, 21, 24, 27];
+        assert_eq!(expected_fizz, actual_fizz);
+
+        let actual_buzz = buzz_output.iter().collect::<Vec<i32>>();
+        let expected_buzz = vec![5, 10, 20, 25];
+        assert_eq!(expected_buzz, actual_buzz);
+
+        let actual_other = other_output.iter().collect::<Vec<i32>>();
+        let expected_other = vec![1, 2, 4, 7, 8, 11, 13, 14, 16, 17, 19, 22, 23, 26, 28, 29];
+        assert_eq!(expected_other, actual_other);
+    }
+
+    #[test]
+    fn fizz_buzz_to_even_odd() {
+        let default_channel_size = 10;
+        let packet_generator = immediate_stream(0..=30);
+
+        let fizz_buzz_elem = ClassifyFizzBuzzElement { id: 0 };
+
+        let mut fizz_buzz_elem_link: ClassifyElementLink<ClassifyFizzBuzzElement> =
```

Comment:
Is the fully specified type needed here because we have two different ClassifyElementLinks?